pandas/core/arrays/datetimelike.py

```diff
@@ -332,6 +332,11 @@ def _validate_frequency(cls, index, freq, **kwargs):
             # Frequency validation is not meaningful for Period Array/Index
             return None
 
+        # DatetimeArray may pass `ambiguous`, nothing else will be accepted
+        #  by cls._generate_range below
```

Comment:
I don't really like this, can we just directly pass ``ambiguous`` as required?

Reply:
We could, but it is much more verbose (BTW the letter "v" on my keyboard is super-sticky, so we really need to stop talking about erbosity).

This would end up looking something like:

```
if is_timedelta64_dtype(index):
    assert ambiguous is None
    on_freq = cls._generate_range(start=index[0], end=None,
                                                        periods=len(index), freq=freq)
else:
    on_freq = cls._generate_range(start=index[0], end=None,
                                                        periods=len(index), freq=freq, ambiguous=ambiguous)
```

Or do you have something else in mind?  I like this less than the way the PR currently does it, but by a small enough margin that I'm fine changing it to get this done.

Reply:
no what I would do is add this to DatetimeIndex. It doesn't totally de-duplicate but is more correct and less hacky here

```
@classmethod
    def _generate_range(cls, start, end, periods, name=None, freq=None,
                        tz=None, normalize=False, ambiguous='raise',
                        closed=None):
        return super(DatetimeIndex, cls)._generate_range(
            start, end, periods, freq,
            tz=tz, normalize=normalize, ambiguous=ambiguous, closed=closed)
```